packages/amplify-util-mock/src/api/lambda-invoke.ts

```diff
@@ -1,39 +1,87 @@
 import { $TSAny, $TSContext } from "amplify-cli-core";
 import { loadLambdaConfig } from '../utils/lambda/load-lambda-config';
-import { BuildType } from 'amplify-function-plugin-interface';
+import { BuildType, FunctionRuntimeLifecycleManager, BuildRequest } from 'amplify-function-plugin-interface';
 import { getInvoker, getBuilder } from 'amplify-category-function';
 import { timeConstrainedInvoker } from '../func';
 import { printer } from 'amplify-prompts';
+import { LambdaTrigger, LambdaTriggerConfig } from "../utils/lambda/find-lambda-triggers";
 
 /**
  * Utility method to invoke the lambda function locally. 
  * Ensures latest function changes are built before invoking it.
  * @param context The CLI context
- * @param functionName Lambda function to invoke locally
+ * @param trigger Lambda trigger to invoke locally
  * @param data Data to be passed to the local lambda function invocation.
  */
-export const invokeLambda = async (context: $TSContext, functionName: string, data: $TSAny): Promise<void> => {
+export const invokeTrigger = async (context: $TSContext, trigger: LambdaTrigger, data: $TSAny): Promise<void> => {
+  let invoker: $TSAny;
+
+  if (trigger?.name) {
+    const functionName = trigger.name;
     const lambdaConfig = await loadLambdaConfig(context, functionName, true);
-      if (!lambdaConfig?.handler) {
-        throw new Error(`Could not parse handler for ${functionName} from cloudformation file`);
-      }
-      // Ensuring latest function changes are built
-      await getBuilder(context, functionName, BuildType.DEV)();
-      const invoker = await getInvoker(context, { resourceName: functionName, handler: lambdaConfig.handler, envVars: lambdaConfig.environment });
-      printer.info('Starting execution...');
-      try {
-        const result = await timeConstrainedInvoker(invoker({ event: data }), context?.input?.options);
-        const stringResult = stringifyResult(result);
-        printer.success('Result:');
-        printer.info(stringResult);
-      } catch (err) {
-        printer.error(`${functionName} failed with the following error:`);
-        printer.info(err);
-      } finally {
-        printer.info('Finished execution.');
-      }
+    if (!lambdaConfig?.handler) {
+      throw new Error(`Could not parse handler for ${functionName} from cloudformation file`);
```

Comment:
will stop commenting related to new Error, but we should use `new AmplifyError` or `new AmplifyFault` everywhere

Reply:
Migrated all the higher level errors to use `AmplifyFault`.